ts/features/idpay/onboarding/components/FailureMessage.tsx

```diff
@@ -20,35 +21,16 @@ type OnboardingFailure = {
 };
 
 const FailureMessage = (props: Props) => {
-  const failures: { [key in OnboardingFailureType]: OnboardingFailure } = {
-    [OnboardingFailureType.NO_PERMISSION]: {
-      picture: "question",
-      title: "Non hai i requisiti per aderire a questa iniziativa",
-      subtitle: "Non puoi richiedere il bonus!"
-    },
-    [OnboardingFailureType.GENERIC]: {
-      picture: "umbrella",
-      title: "C’è un problema sui nostri sistemi",
-      subtitle:
-        "Ci scusiamo per il disagio.\nTi invitiamo a riprovare più tardi."
-    },
-    [OnboardingFailureType.ALREADY_COMPLETED]: {
-      picture: "inProgress",
-      title: "Onboarding concluso",
-      subtitle: "Hai già fatto l'onboarding per questa iniziativa"
-    },
-    [OnboardingFailureType.ONBOARDING_KO]: {
-      picture: "question",
-      title: "Onboarding fallito",
-      subtitle: "L'adesione non è andata a buon fine"
-    }
-  };
-
-  const failure = failures[props.type];
-
-  if (failure === undefined) {
+  if (OnboardingFailureType[props.type] === undefined) {
     return null;
   }
+  const failure: OnboardingFailure = {
+    picture: I18n.t(
+      `idpay.onboarding.failure.${props.type}.picture`
+    ) as IOPictogramType,
```

Comment:
Does the `pictogram` image depends on the localization? If not, could a `Record<OnboardingFailureType, IOPictogramType>` be a more robust (and simpler) solution in this case?